src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/VirtualDocumentBase.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServer.ContainedLanguage
+{
+    public abstract class VirtualDocumentBase<T> : VirtualDocument where T : VirtualDocumentSnapshot
+    {
+        private T _currentSnapshot;
+        private long? _hostDocumentSyncVersion;
+
+        protected VirtualDocumentBase(Uri uri, ITextBuffer textBuffer)
+        {
+            if (uri is null)
+            {
+                throw new ArgumentNullException(nameof(uri));
+            }
+
+            if (textBuffer is null)
+            {
+                throw new ArgumentNullException(nameof(textBuffer));
+            }
+
+            Uri = uri;
```

Comment:
Does the razor team prefer this code structure to the 

Uri = uri ?? throw new ArgumentNullException(nameof(uri));

structure in their code?

Reply:
We use this structure in more places, but I've seen the `??` method used a couple times and I don't think anyone complained.

Reply:
I left this as is to be consistent because that's what Taylor used in most of this code. Sounds like he still has a preference for this style.

---
In reply to: [450335794](https://github.com/dotnet/aspnetcore-tooling/pull/2174#discussion_r450335794) [](ancestors = 450335794)

Reply:
Yeah, I left it as this to follow the style you use. 

---
In reply to: [450341271](https://github.com/dotnet/aspnetcore-tooling/pull/2174#discussion_r450341271) [](ancestors = 450341271)